src/openms/source/FORMAT/MzQCFile.cpp

```diff
@@ -0,0 +1,199 @@
+                       const MSExperiment & exp,
+                       const String & contactName,
+                       const String & contactAddress,
+                       const String & description,
+                       const String & label) const
+  {
+    using json = nlohmann::ordered_json;
+    json out;
+    // required: creationDate, version
+    DateTime currentTime = DateTime::now();
+    out["MzQC"]["creationDate"] = currentTime.toString();
+    out["MzQC"]["version"] = "1.0.0";
+
+    // optional: contactName, contactAddress, description
+    if (!contactName.empty()) 
+    {
+      out["MzQC"]["contactName"] = contactName;
+    }
+    if (!contactAddress.empty()) 
+    {
+      out["MzQC"]["contactAddress"] = contactAddress;
+    }
+    if (!description.empty()) 
+    {
+      out["MzQC"]["description"] = description;
+    }
+
+    out["MzQC"]["runQualities"]["metadata"]["label"] = label;
+    out["MzQC"]["runQualities"]["metadata"]["inputFiles"] = {
+        {
+          {"location", File::absolutePath(inputFileName)},
+          {"name", File::basename(inputFileName)},
+          {"fileFormat",
+          {
+            {"accession", "MS:10000584"},
+            {"name", "mzML format"}
+          }
+          }
+        },
+          "fileProperties",
+          {
+            {
+              {"accession", "MS:1000747"},
+              {"name", "completion time"},
+              {"value", String(exp.getDateTime().getDate() + "T" + exp.getDateTime().getTime())}
+            },
+            {
+              {"accession", "MS:1000569"},
+              {"name", "SHA-1"},
+              {"value", String(FileHandler::computeFileHash(inputFileName))}
+            },
+            {
+              {"accession", "MS:1000031"},
+              {"name", "instrument model"},
+              {"value", String(exp.getInstrument().getName())}
+            }
+          }
+    };
+
+    VersionInfo::VersionDetails version = VersionInfo::getVersionStruct();
+    out["MzQC"]["runQualities"]["metadata"]["analysisSoftware"] = { // todo
+        {
+          {"accession", "MS:1009001" }, // create new qc-cv for QCCalculator: MS:1009001 quality control metrics generating software
+          {"name", "QCCalculator"},
+          {"version", String(version.version_major)+"."+String(version.version_minor)+"."+String(version.version_patch)},
+          {"uri", "https://www.openms.de"}
+        }
+    };
+
+    out["MzQC"]["runQualities"]["qualityMetrics"] = {};
+    
+    out["MzQC"]["controlledVocabularies"] = {
+        {
+          {"name", "Proteomics Standards Initiative Quality Control Ontology"},
+          {"uri", "https://github.com/HUPO-PSI/mzQC/blob/master/cv/qc-cv.obo"},
+          {"version", "0.1.2"},
+        },
+        {
+          {"name", "Proteomics Standards Initiative Mass Spectrometry Ontology"},
+          {"uri", "https://github.com/HUPO-PSI/psi-ms-CV/blob/master/psi-ms.obo"},
+          {"version", "4.1.49"}
```

Comment:
We could pull out all version numbers and put them in the header file as:
e.g,
constexpr char psi_ms_obo_version[] = "4.1.49";
so we have all version numbers at one place.
Not sure if worth it

Reply:
personally i would leave it as it is since we have to edit everything here anyways when updating the cv